doc/src/modules/physics/mechanics/examples/multi_degree_freedom_holonomic_system.rst

```diff
@@ -0,0 +1,86 @@
+``dynamicsymbols`` needed to describe the system as shown in the above diagram.
+In this case, the generalized coordinates :math:`q_1` represent lateral distance of block from wall,
+:math:`q_2` represents ngle of the compound pendulum from vertical, :math:`q_3`  represents angle of the simple
+pendulum from the compound pendulum. The generalized speeds :math:`u_1` represents lateral speed of block,
+:math:`u_2` represents lateral speed of compound pendulum and :math:`u_3` represents angular speed of C relative to B.
+
+We also create some ``symbols`` to represent the length and
+mass of the pendulum, as well as gravity and others. ::
+
+    >>> from sympy import zeros, symbols
+    >>> from sympy.physics.mechanics import *
+    >>> q1, q2, q3, u1, u2, u3 = dynamicsymbols('q1, q2, q3, u1, u2, u3')
+    >>> l, k, c, g, kT = symbols('l, k, c, g, kT')
+    >>> ma, mb, mc, IBzz= symbols('ma, mb, mc, IBzz')
+
+Next, we create the bodies and connect them using joints to establish the
+kinematics. ::
+
+    >>> wall = Body('N')
+    >>> block = Body('A', mass=ma)
+    >>> IB = inertia(block.frame, 0, 0, IBzz)
+    >>> compound_pend = Body('B', mass=mb, central_inertia=IB)
+    >>> simple_pend = Body('C', mass=mc)
+
+    >>> bodies = (wall, block, compound_pend, simple_pend)
+
+    >>> slider = PrismaticJoint('J1', wall, block, coordinates=q1, speeds=u1)
+    >>> rev1 = PinJoint('J2', block, compound_pend, coordinates=q2, speeds=u2,
+    ...                 child_axis=compound_pend.z, child_joint_pos=l*2/3*compound_pend.y,
+    ...                 parent_axis=block.z)
+    >>> rev2 = PinJoint('J3', compound_pend, simple_pend, coordinates=q3, speeds=u3,
+    ...                 child_axis=simple_pend.z, parent_joint_pos=-l/3*compound_pend.y,
+    ...                 parent_axis=compound_pend.z, child_joint_pos=l*simple_pend.y)
+
+    >>> joints = (slider, rev1, rev2)
+
+Now we can apply loads(forces and torques) to the bodies, gravity acts on all bodies,
+a linear spring and damper act on block and wall, a rotational linear spring acts on C relative to B
+specified torque T acts on compound_pend and block, specified force F acts on block. ::
+
+    >>> F, T = dynamicsymbols('F, T')
+    >>> block.apply_force(F*block.x)
+    >>> block.apply_force(-k*q1*block.x, reaction_body=wall)
+    >>> block.apply_force(-c*u1*block.x, reaction_body=wall)
+    >>> compound_pend.apply_torque(T*compound_pend.z, reaction_body=block)
+    >>> simple_pend.apply_torque(kT*q3*simple_pend.z, reaction_body=compound_pend)
+    >>> block.apply_force(-wall.y*block.mass*g)
+    >>> compound_pend.apply_force(-wall.y*compound_pend.mass*g)
+    >>> simple_pend.apply_force(-wall.y*simple_pend.mass*g)
+
+With the problem setup, the equations of motion can be generated using the
+``JointsMethod`` class with KanesMethod in backend. ::
+
+    >>> method = JointsMethod(wall, slider, rev1, rev2)
+    >>> fr, frstar = method.form_eoms()
```

Comment:
I don't think this should return fr and frstar. If it returns anything it should return `fr + frstar`.

Reply:
But I think user would expect the same result as given by `KanesMethod`.